test_rmw_implementation/test/test_create_destroy_node.cpp

```diff
@@ -131,3 +131,41 @@ TEST_F(CLASSNAME(TestNodeConstructionDestruction, RMW_IMPLEMENTATION), create_an
   ASSERT_NE(nullptr, node) << rmw_get_error_string().str;
   EXPECT_EQ(RMW_RET_OK, rmw_destroy_node(node)) << rmw_get_error_string().str;
 }
+
+class CLASSNAME (TestNodeConstructionDestructionLocalHost, RMW_IMPLEMENTATION) : public ::testing::Test
+{
+protected:
+  void SetUp() override
+  {
+    options = rmw_get_zero_initialized_init_options();
+    rmw_ret_t ret = rmw_init_options_init(&options, rcutils_get_default_allocator());
+    ASSERT_EQ(RMW_RET_OK, ret) << rcutils_get_error_string().str;
+    options.enclave = rcutils_strdup("/", rcutils_get_default_allocator());
+    ASSERT_STREQ("/", options.enclave);
+    options.localhost_only = RMW_LOCALHOST_ONLY_ENABLED;
+    context = rmw_get_zero_initialized_context();
+    ret = rmw_init(&options, &context);
+    ASSERT_EQ(RMW_RET_OK, ret) << rcutils_get_error_string().str;
+  }
+
+  void TearDown() override
+  {
+    rmw_ret_t ret = rmw_shutdown(&context);
+    EXPECT_EQ(RMW_RET_OK, ret) << rcutils_get_error_string().str;
+    ret = rmw_context_fini(&context);
+    EXPECT_EQ(RMW_RET_OK, ret) << rcutils_get_error_string().str;
+    ret = rmw_init_options_fini(&options);
+    EXPECT_EQ(RMW_RET_OK, ret) << rcutils_get_error_string().str;
+  }
+
+  rmw_init_options_t options;
+  rmw_context_t context;
+};
+
+TEST_F(CLASSNAME(TestNodeConstructionDestructionLocalHost, RMW_IMPLEMENTATION), create_and_destroy) {
+  const char * const node_name = "my_node";
```

Comment:
@Lobotuerk nit: 
```suggestion
  constexpr char node_name[] = "/my_ns";
```

Same below.

Reply:
Does not seem to work like that, 
```
42: /home/lobo/rmw_tests/ros2_rolling/src/ros2/rmw_implementation/test_rmw_implementation/test/test_create_destroy_node.cpp:173: Failure
42: Expected: (nullptr) != (node), actual: (nullptr) vs NULL
42: invalid node name: node name must not contain characters other than alphanumerics or '_', at /home/lobo/rmw_tests/ros2_rolling/src/ros2/rmw_fastrtps/rmw_fastrtps_shared_cpp/src/rmw_node.cpp:58
```

Reply:
Oof, I meant you to use `constexpr`, not the string ! My bad, it wasn't clear.

Reply:
Should i change the other usages of `const char * const` on other parts of the file? I kept them for consistency